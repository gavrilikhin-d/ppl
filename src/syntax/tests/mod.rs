use crate::test_compilation_result;

test_compilation_result!(consume_greater);
test_compilation_result!(escaped_id);
test_compilation_result!(multiple_errors);
