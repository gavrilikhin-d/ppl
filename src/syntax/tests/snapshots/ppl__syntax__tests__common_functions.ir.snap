---
source: src/syntax/tests/mod.rs
expression: ir
---
; ModuleID = 'common_functions'
source_filename = "common_functions.ppl"

%String = type { ptr }

@0 = private unnamed_addr constant [4 x i8] c"foo\00", align 1

define void @foo() !dbg !3 {
  %1 = call %String @string_from_c_string_and_length(ptr @0, i64 3), !dbg !7
  call void @"println <:String>"(%String %1), !dbg !7
  br label %return, !dbg !7

return:                                           ; preds = %0
  ret void
}

declare void @"println <:String>"(%String)

declare %String @string_from_c_string_and_length(ptr, i64)

define void @common_functions.execute() !dbg !8 {
  call void @foo(), !dbg !9
  br label %return, !dbg !9

return:                                           ; preds = %0
  ret void
}

!llvm.module.flags = !{!0}
!llvm.dbg.cu = !{!1}

!0 = !{i32 2, !"Debug Info Version", i32 3}
!1 = distinct !DICompileUnit(language: DW_LANG_C, file: !2, producer: "ppl", isOptimized: false, runtimeVersion: 0, emissionKind: FullDebug, splitDebugInlining: false, sysroot: "/")
!2 = !DIFile(filename: "common_functions.ppl", directory: ".")
!3 = distinct !DISubprogram(name: "foo", linkageName: "foo", scope: !2, file: !2, type: !4, spFlags: DISPFlagDefinition, unit: !1)
!4 = !DISubroutineType(types: !5)
!5 = !{!6}
!6 = !DIBasicType(name: "i32", size: 32, encoding: DW_ATE_signed)
!7 = !DILocation(line: 0, column: 18, scope: !3)
!8 = distinct !DISubprogram(name: "common_functions.execute", linkageName: "common_functions.execute", scope: !2, file: !2, line: 1, type: !4, spFlags: DISPFlagDefinition, unit: !1)
!9 = !DILocation(line: 1, scope: !8)
